scripts/BBoxManage.sh

```diff
@@ -0,0 +1,474 @@
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function unmountusb () {
+  # Unmount usb
+  if [ $DEBUG == 1 ]; then
+    echo "Unmounting USB media"
+  fi
+
+  pumount /media/usb0 2>&1 | logger -t $(basename $0)
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doshutdown () {
+  # Shutdown
+  if [ $DEBUG == 1 ]; then
+    echo "Shutting down"
+  fi
+
+  /sbin/shutdown now 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doreboot () {
+  # Reboot
+  if [ $DEBUG == 1 ]; then
+    echo "Rebooting"
+  fi
+
+  /sbin/reboot 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function restart_hostapd () {
+  # Restart the hostapd service
+  if [ $DEBUG == 1 ]; then
+    echo "Restarting hostapd"
+  fi
+
+  /etc/init.d/hostapd restart
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function set_password () {
+  if [[ -z "${val// }" ]]; then
+    echo 'Missing password value'
+    exit 1;
+  fi
+
+  backup_password_config
+
+  # Update the password property in the basicauth configuration file
+  if [ $DEBUG == 1 ]; then
+    echo "Updating password to '$val'"
+  fi
+
+  local new_hash=`echo $val | openssl passwd -apr1 -salt $PASSWORD_SALT -stdin`
+  echo "biblebox:$new_hash" > $PASSWORD_CONFIG 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    reload_nginx
+  else
+    failure
+  fi
+}
+
+function get_hostname () {
+  local hostname=`grep 'bbox_server_name=' $NGINX_CONFIG`
+  local idx=`expr index "$hostname" =`
+
+  web_hostname="${hostname:$idx:${#hostname}-$idx}"
+
+  echo $web_hostname;
+}
+
+function set_hostname () {
+  if [[ -z "${val// }" ]]; then
+    echo 'Missing hostname value'
+    exit 1;
+  fi
+
+  backup_nginx_config
+
+  # Update the hostname in the nginx config
+  if [ $DEBUG == 1 ]; then
+    echo "Updating hostname to '$val'"
+  fi
+
+  get_hostname > /dev/null #suppress output
+
+  sed -i "s/$web_hostname/$val/g" $NGINX_CONFIG 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    reload_nginx
+  else
+    failure
+  fi
+}
+
+function get_channel () {
+  local channel=`grep '^channel=' $HOSTAPD_CONFIG`
+  if [[ $channel == channel=\"* ]]; then
+    echo ${channel:9:${#channel}-10};
+  else
+    echo ${channel:8:${#channel}-8};
+  fi
+}
+
+function set_channel () {
+  if [[ -z "${val// }" ]]; then
+    echo 'Missing channel value'
+    exit 1;
+  fi
+
+  backup_hostapd_config
+
+  # Update the channel property in the hostapd configuration file
+  if [ $DEBUG == 1 ]; then
+    echo "Updating channel to '$val'"
+  fi
+
+  sed -i "s/^channel=.*/channel=$val/g" $HOSTAPD_CONFIG 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    restart_hostapd
+  else
+    failure
+  fi
+}
+
+function get_ssid () {
```

Comment:
For the ssid property in /etc/hostapd/hostapd.conf, I think it can be either quoted or unquoted, which is why there is that if statement.